ansible_collections/arista/cvp/plugins/module_utils/tag_tools.py

```diff
@@ -0,0 +1,238 @@
+                                                        remove=True)
+
+            if 'interface_tags' in tag_type:
+                element_type = "ELEMENT_TYPE_INTERFACE"
+                for intf_tags in per_device['interface_tags']:
+                    interface_id = intf_tags['interface']
+                    for tag in intf_tags['tags']:
+                        tag_name = tag['name']
+                        tag_val = tag['value']
+                        if mode == 'create':
+                            self.__cv_client.api.tag_config(element_type, workspace_id,
+                                                            tag_name, tag_val)
+                        if state == 'assign':
+                            if auto_create:
+                                self.__cv_client.api.tag_config(element_type, workspace_id,
+                                                                tag_name, tag_val)
+                            self.__cv_client.api.tag_assignment_config(element_type,
+                                                                       workspace_id,
+                                                                       tag_name,
+                                                                       tag_val,
+                                                                       device_id,
+                                                                       interface_id)
+                        if state == 'unassign':
+                            self.__cv_client.api.tag_assignment_config(element_type,
+                                                                       workspace_id,
+                                                                       tag_name,
+                                                                       tag_val,
+                                                                       device_id,
+                                                                       interface_id,
+                                                                       remove=True)
+                        if mode == 'delete':
+                            self.__cv_client.api.tag_config(element_type, workspace_id,
+                                                            tag_name, tag_val,
+                                                            remove=True)
+
+        # Start build
+        request = 'REQUEST_START_BUILD'
+        reques_id = 'b1'
+        description = 'testing cvprac build'
+        self.__cv_client.api.workspace_config(workspace_id=workspace_id,
+                                              display_name=workspace_name,
+                                              description=description, request=request,
+                                              request_id=reques_id)
+
+        # Check workspace build status and proceed only after it finishes building
+        # XXX: timeout? ansible timeout!
```

Comment:
Strange comment here

Reply:
More for my notes, as a reminder of future enhancements.

Reply:
Ok that's fine to have small comments like that, but maybe try to make them more understandable for other people. 
Something like: 
```
TODO: Handle better timeout here
```